### PR TITLE
feat(netlify-cms-widget-markdown): upgrade rehype/remark/unified related dependencies

### PR DESCRIPTION
**Pull request by alexandernanberg**
**Summary**

Upgrade rehype/remark/unified related dependencies. The motivation behind this is to stay up-to-date so we still receive path updates and new features.

**Test plan**
N/A

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 610a6164afee0d57464cffb9bab4e47e18edd406

https://deploy-preview-1885--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 610a6164afee0d57464cffb9bab4e47e18edd406

https://deploy-preview-1885--cms-demo.netlify.com

**Comment by erquhart** [comment]
Thanks! No breaking changes in any of those dependency updates?

**Comment by alexandernanberg** [comment]
I dug through all the release notes and didn't find anything that should affect us, so I've not found anything other than what I noted above with the links. Not sure if that should be treated as a breaking change or not 🤔

**Comment by erquhart** [comment]
Hmm yeah that change is breaking: the original markdown syntax did support an optional space in link references ([source](https://daringfireball.net/projects/markdown/syntax#link)), but Commonmark diverged on that point, hence this change ([source](https://github.com/remarkjs/remark/pull/355)).

As silly as it is, we'll need to wait until 3.0 to update remark-parse. Really wish we still a 0.x project 🙄.

Leave it open, we'll circle back 👍

File: packages/netlify-cms-widget-markdown/src/__tests__/renderer.spec.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ Text with **bold** & _em_ elements
     describe('Links', () => {
       it('should render links', () => {
         const value = `
-I get 10 times more traffic from [Google] [1] than from [Yahoo] [2] or [MSN] [3].
+I get 10 times more traffic from [Google][1] than from [Yahoo][2] or [MSN][3].
```
[the remark this answers is not in the document]
Comment (erquhart) [reply packages/netlify-cms-widget-markdown/src/__tests__/renderer.spec.js]: They're both valid, but it isn't a part of commonmark or gfm, and neither format works at all in the cms currently anyway. No worries.

**Comment by erquhart** [comment]
Closing this as I'm about to put up a PR that handles these updates together with a lot of others.

**Pull request closed without merging**